march_gait_selection/src/march_gait_selection/dynamic_gaits/balance_gait.py

```diff
@@ -44,34 +51,39 @@ def capture_point_cb(self, msg, leg_name):
         :param leg_name: The name of corresponding move group
         """
         self._latest_capture_point_msg_time[leg_name] = msg.header.stamp
-        self._capture_point_position[leg_name] = msg.pose.position
+        self._capture_point_pose[leg_name] = msg.pose
 
     def calculate_trajectory(self, leg_name):
         """Calculate the trajectory using moveit and return as a subgait msg format.
 
         :param leg_name: The name of the used move group
 
         :return:
-            A populated subgait message
+            A joint trajectory which can be used in a subgait or subgait msg
         """
-        return self.random_subgait()
+        if self._capture_point_pose[leg_name] is None:
+            rospy.logwarn('No messages received from the capture point topic of {lg}'.format(lg=leg_name))
+            return None
 
-    def random_subgait(self):
-        """Create random trajectory using the moveit motion planner.
+        if self._old_capture_point_msg_time[leg_name]:
+            if self._old_capture_point_msg_time[leg_name] == self._latest_capture_point_msg_time[leg_name]:
+                rospy.logwarn('No new capture point messages received from cp topic of {lg}'.format(lg=leg_name))
+                return None
 
-        :return:
-            A populated subgait message
-        """
-        self._move_group['left_leg'].set_random_target()
-        trajectory_msg = self._move_group['left_leg'].plan()
-        return self.to_subgait_msg_('Random moveit subgait', trajectory_msg.joint_trajectory)
+        self._old_capture_point_msg_time[leg_name] = self._latest_capture_point_msg_time[leg_name]
+
+        _pose = self._capture_point_pose[leg_name]
+        _end_effector = self._end_effectors[leg_name]
+        self._move_group[leg_name].set_joint_value_target(_pose, _end_effector, True)
```

Comment:
local variables mustn't use a preceding underscore. This confused me, because I thought you were missing `self.`
```suggestion
        pose = self._capture_point_pose[leg_name]
        end_effector = self._end_effectors[leg_name]
        self._move_group[leg_name].set_joint_value_target(pose, end_effector, True)
```

Reply:
I will fix this